crates/rpc/rpc/src/debug.rs

```diff
@@ -257,7 +257,19 @@ where
                         return Ok(frame.into())
                     }
                     GethDebugBuiltInTracerType::PreStateTracer => {
-                        Err(EthApiError::Unsupported("pre state tracer currently unsupported."))
+                        let prestate_config = tracer_config
+                            .into_pre_state_config()
+                            .map_err(|_| EthApiError::InvalidTracerConfig)?;
+                        let mut inspector = TracingInspector::new(TracingInspectorConfig::from_geth_config(&config));
+
+                        let (res, _, db) = self
+                            .inner
+                            .eth_api
+                            .inspect_call_at_and_return_state(call, at, overrides, &mut inspector)
```

Comment:
now that #3914 is merged we need to rebase the pr and make a small API change here, where we need to move all that logic inside a `spawn_with_call_at` call, similar to:

https://github.com/paradigmxyz/reth/blob/b5a44aeb3764265eb26b5c9794371710513f4875/crates/rpc/rpc/src/debug.rs#L308-L312

which accepts a callback in which we need to inspect and then also create the frame